api/app/tests/apps/services/domain/auto_create_children.py

```diff
@@ -0,0 +1,145 @@
+    def setUp(self):
+        self.category_container_is_vol = Category.objects.get(slug='container-is-vol', parent__isnull=False)
+        self.extra_properties = [{
+            "id": "extra_container",
+            "label": "Container(s)",
+            "answer": [
+                {"id": "ID001", "type": "Rest", "description": "Restafval container"},
+                {"id": "ID002", "type": "Plastic", "description": "Plasticafval container"},
+                {"id": "ID003", "type": "Papier", "description": "Papierafval container"},
+                {"id": "ID0004", "type": "Unknown", "description": "Unknown container"},
+            ],
+            "category_url": "/signals/v1/public/terms/categories/afval/sub_categories/container-is-vol"
+        }]
+
+    @mock.patch(
+        'signals.apps.services.domain.auto_create_children.AutoCreateChildrenService._get_container_location',
+        autospec=True
+    )
+    def test_create_children(self, mocked):
+        extra_properties = copy.deepcopy(self.extra_properties)
+        signal = SignalFactory.create(extra_properties=extra_properties,
+                                      category_assignment__category=self.category_container_is_vol)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        mocked.return_value = signal.location.geometrie
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertTrue(signal.is_parent)
+        self.assertEqual(signal.children.count(), 4)
+
+    def test_no_containers_selected(self):
+        signal = SignalFactory.create(extra_properties=None,
+                                      category_assignment__category=self.category_container_is_vol)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+    def test_one_container_selected(self):
+        extra_properties = copy.deepcopy(self.extra_properties)
+        extra_properties[0]['answer'] = [extra_properties[0]['answer'][0]]
+
+        signal = SignalFactory.create(extra_properties=None,
+                                      category_assignment__category=self.category_container_is_vol)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+    @mock.patch(
+        'signals.apps.services.domain.auto_create_children.AutoCreateChildrenService._get_container_location',
+        autospec=True
+    )
+    def test_multiple_extra_properties_for_containers(self, mocked):
```

Comment:
Is there also a maximum we should be checking? (For this feature or when accepting new complaints?)